ports/krb5/portfile.cmake

```
@@ -1,24 +1,75 @@
-vcpkg_download_distfile(ARCHIVE
-    URLS "https://kerberos.org/dist/krb5/1.21/krb5-${VERSION}.tar.gz"
-    FILENAME "krb5-${VERSION}.tar.gz"
-    SHA512 4e09296b412383d53872661718dbfaa90201e0d85f69db48e57a8d4bd73c95a90c7ec7b6f0f325f6bc967f8d203b256b071c0191facf080aca0e2caec5d0ac49
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO krb5/krb5
+    REF krb5-${VERSION}-final
+    SHA512 184ef8645d7e17f30a8e3d4005364424d2095b3d0c96f26ecef0c2dd2f3a096a0dd40558ed113121483717e44f6af41e71be0e5e079c76a205535d0c11a2ea34
+    HEAD_REF master
+    PATCHES
+        fix_relative_path.patch
 )
 
-vcpkg_extract_source_archive(
-    SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
-)
-
-vcpkg_configure_make(
-    SOURCE_PATH "${SOURCE_PATH}/src"
-    AUTOCONFIG
-    OPTIONS
-        "CFLAGS=-fcommon \$CFLAGS"
-)
-
-vcpkg_install_make()
+if (VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_acquire_msys(MSYS_ROOT PACKAGES)
+    vcpkg_add_to_path("${MSYS_ROOT}/usr/bin")
+    vcpkg_find_acquire_program(PERL)
+    get_filename_component(PERL_PATH ${PERL} DIRECTORY)
+    vcpkg_add_to_path("${PERL_PATH}")
+    vcpkg_build_nmake(
+        SOURCE_PATH "${SOURCE_PATH}/src"
+        PROJECT_NAME Makefile.in
+        TARGET prep-windows
+    )
+    file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/ DESTINATION ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET})
+    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug")
+    vcpkg_install_nmake(
+        SOURCE_PATH "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}"
+        PROJECT_NAME "Makefile"
+        OPTIONS
+            "NO_LEASH=1"
+        OPTIONS_RELEASE
+            "KRB_INSTALL_DIR=${CURRENT_PACKAGES_DIR}"
+        OPTIONS_DEBUG
+            "KRB_INSTALL_DIR=${CURRENT_PACKAGES_DIR}/debug"
+    )
+    set(tools 
+        ccapiserver
+        gss-client
+        gss-server
+        kcpytkt
+        kdeltkt
+        kdestroy
+        kfwcpcc
+        kinit
+        klist
+        kpasswd
+        kswitch
+        kvno
+        mit2ms
+        ms2mit
+    )
+    vcpkg_copy_tools(
+        SEARCH_DIR "${CURRENT_PACKAGES_DIR}/bin"
+        DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin"
+        TOOL_NAMES ${tools}
```

As a general pattern, I would prefer to see the TOOL_NAMES first.
```suggestion
        TOOL_NAMES ${tools}
        DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin"
```

---

destination is also the default?

---

I would suggest to leave the explicit argument:
In this port, it ensures that windows (nmake, vcpkg_copy_tools) install to the same location as non-windows (vcpkg_configure/install_make).

---

ok.